src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.TryReadStatusFile.cs

```diff
@@ -149,21 +152,63 @@ internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus
             return true;
         }
 
-        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, [NotNullWhen(true)] out string? fileContents)
+        private static bool TryReadFile(string path, [NotNullWhen(true)] out string? contents)
         {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+
+            if (_encoding == null || _decoder == null)
+            {
+                _encoding = Encoding.UTF8;
+                _decoder = _encoding.GetDecoder();
+            }
+
+            char[]? chars = null;
+            byte[]? bytes = null;
+            const int size = 4096;
```

Comment:
This seems to be assuming that all files read will always be <= 4096 bytes?

Reply:
Updated with re-renting of buffer when file has more content.